client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -57,16 +92,26 @@ export class SimpleSiteRenameForm extends Component {
 	};
 
 	onFieldChange = event => {
-		this.setState( { domainFieldValue: get( event, 'target.value' ) } );
+		const domainFieldValue = get( event, 'target.value' );
+		const maybeDomainFieldError = this.state.domainFieldError && {
```

Comment:
This line is hard for me to understand - could you simplify the logic here so that it's easier to grasp what you're trying to achieve?
After mulling it over, I understand that you're trying to update the validation error only when we're already showing one, but it's hard to get to that based on this line.
I think the naming of the variable is just to ambiguous. Maybe something like:
```
const { domainFieldError } = this.state;
const errorNeedsUpdating = ! isEmpty( domainFieldError );
this.setState( {
	domainFieldValue,
	domainFieldError: errorNeedsUpdating ? this.getDomainValidationMessage( domainFieldValue ) : domainFieldError,
} );
```
It's a bit more verbose, but hopefully easier to understand. I'm of course open to other suggestions how this can be written, this is just an example.